pkg/dns/dns_proxy_handler.go

```diff
@@ -109,7 +110,9 @@ func getUpdateCacheFunc(log logr.Logger, cache *DNSCache) func(lookupTime time.T
 	return func(lookupTime time.Time, response *dnsgo.Msg) {
 		if response.Response && response.Rcode == dnsgo.RcodeSuccess {
 			scopedLog := log.WithValues(reqIdLogField, response.Id)
-			if err := cache.Update(lookupTime, response); err != nil {
+			qname := strings.ToLower(response.Question[0].Name)
```

Comment:
must be checked e.g. `if len(response.Question) > 0`

Reply:
Should be fixed now; can you check if this is right?